base/irrationals.jl

```diff
@@ -221,6 +221,15 @@ end
 log(::Irrational{:e}) = 1 # use 1 to correctly promote expressions like log(x)/log(e)
 log(::Irrational{:e}, x::Number) = log(x)
 
+# These methods should go to special/trig.jl, but Irrational is not yet defined when the
+# file is loaded.
+sinpi(x::Irrational) = sinpi(float(x))
+cospi(x::Irrational) = cospi(float(x))
+sinc(x::Irrational) = sinc(float(x))
+cosc(x::Irrational) = cosc(float(x))
+sinc(z::Complex{<:Irrational}) = sinc(float(z))
+cosc(z::Complex{<:Irrational}) = cosc(float(z))
```

Comment:
Because other functions already works with irrationals! :-)  These are the only functions that I spotted that need to be explicitly defined for irrationals.  The problem is that methods of `sinpi` and `cospi` with argument `T<:Real` were probably implemented with `Integer` and `Rational` in mind, while `Irrational` are in a certain sense closer to `AbstractFloat`.